src/libexpr/eval.cc

```diff
@@ -2472,30 +2473,40 @@ std::ostream & operator << (std::ostream & str, const ExternalValueBase & v) {
 
 EvalSettings::EvalSettings()
 {
-    auto var = getEnv("NIX_PATH");
-    if (var) nixPath = parseNixPath(*var);
 }
 
+/* impure => NIX_PATH or a default path
+ * restrict-eval => NIX_PATH
+ * pure-eval => empty
+ */
 Strings EvalSettings::getDefaultNixPath()
 {
-    Strings res;
-    auto add = [&](const Path & p, const std::string & s = std::string()) {
-        if (pathExists(p)) {
-            if (s.empty()) {
-                res.push_back(p);
-            } else {
-                res.push_back(s + "=" + p);
+    if (pureEval)
+        return {};
+
+    auto var = getEnv("NIX_PATH");
+    if (var) {
+        return parseNixPath(*var);
+    } else {
+        Strings res;
+        auto add = [&](const Path & p, const std::string & s = std::string()) {
```

Comment:
Using 

```suggestion
        auto add = [&](const Path & p, const std::optional<std::string> & s = std::nullopt) {
```

would be cleaner. But since that code already existed before, feel free to ignore this for now